leshan-core/src/main/java/org/eclipse/leshan/core/LwM2m.java

```diff
@@ -91,35 +91,35 @@ public boolean equals(Object obj) {
      */
     public class Version implements Comparable<Version> {
 
-        public static Version V1_0 = new Version("1.0");
-        public static final Version MAX = new Version(Short.MAX_VALUE, Short.MIN_VALUE);
+        public static final Version V1_0 = new Version("1.0");
+        public static final Version MAX = new Version(Short.MAX_VALUE, Short.MAX_VALUE);
 
-        protected final Short major;
-        protected final Short minor;
+        protected final short major;
+        protected final short minor;
 
         public Version(int major, int minor) {
-            this.major = (short) major;
-            this.minor = (short) minor;
+            this((short) major, (short) minor);
+            if (this.major != major) {
+                throw new ArithmeticException("short overflow");
+            }
+            if (this.minor != minor) {
+                throw new ArithmeticException("short overflow");
```

Comment:
Maybe the error message could looks like the one in `validate()` method ? 
maybe something like : 
- `"version (%d.%d) major part (%d) is not a valid short"`
- `"version (%d.%d) minor part (%d) is not a valid short"`